example/helloWorld/src/helloworld.cpp

```diff
@@ -0,0 +1,177 @@
+/**
+ * \file
+ * Copyright 2014-2015 Benjamin Worpitz
+ *
+ * This file is part of alpaka.
+ *
+ * alpaka is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * alpaka is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with alpaka.
+ * If not, see <http://www.gnu.org/licenses/>.
+ *
+ * \author Erik Zenker
+ *
+ */
+
+// STL
+#include <iostream>
+
+// Alpaka
+#include <alpaka/alpaka.hpp>
+
+
+/**
+ * Helper functions that calculates the global thread id depending
+ * on the number of threads set by the workdiv.
+ *
+ */
+template <typename T_Acc>
+size_t globalThreadIdx(T_Acc const &acc){
+    auto threadsExtent = alpaka::workdiv::getWorkDiv<alpaka::Grid, alpaka::Threads>(acc);
+    auto threadIdx = alpaka::idx::getIdx<alpaka::Grid, alpaka::Threads>(acc);
+    
+    auto globalThreadIdx =
+	threadIdx[0]
+	+ threadIdx[1] * threadsExtent[0]
+	+ threadIdx[2] * threadsExtent[0] * threadsExtent[1];
+
+    return globalThreadIdx;
+}
+
+/**
+ * Kernel
+ *
+ * Prints "[tid] Hello World" where tid is
+ * the global thread number.
+ *
+ */
+struct HelloWorldKernel {
+    template <typename T_Acc>
+    ALPAKA_FN_ACC void operator()( T_Acc const &acc) const {
+	printf("[%u] Hello World\n", static_cast<unsigned>(globalThreadIdx(acc)));
+	
+    }
+
+};
+
+int main() {
+
+    /**
+     * Define accelerator types
+     *
+     * It is possible to choose from a set of accelerators
+     * that are defined in the alpaka::acc namespace e.g.:
+     * - AccGpuCudaRt
+     * - AccCpuThreads
+     * - AccCpuFibers
+     * - AccCpuOmp2Threads
+     * - AccCpuOmp2Blocks
+     * - AccCpuOmp4
+     * - AccCpuSerial
+     *
+     * Each accelerator has strengths and weaknesses. Therefore, 
+     * they need to be choosen carefully depending on the actual
+     * use case. Furthermore, some accelerators only support a
+     * particular workdiv, but workdiv can also be generated
+     * automatically.
+     *
+     */
+    using Dim     = alpaka::dim::DimInt<3>;  
+    using Size    = std::size_t;
+    using Host    = alpaka::acc::AccCpuSerial<Dim, Size>;
+    using Acc     = alpaka::acc::AccCpuSerial<Dim, Size>;
+    using Stream  = alpaka::stream::StreamCpuSync;
+    using DevAcc  = alpaka::dev::Dev<Acc>;
+    using DevHost = alpaka::dev::DevCpu;
```

Comment:
For consistency I would prefer `alpaka::dev::Dev<Host>` even though it maps to `DevCpu` in the end.